testflinger_agent/agent.py

```diff
@@ -30,8 +30,17 @@ def __init__(self, client):
         self.client = client
         self._state = multiprocessing.Array("c", 16)
         self.set_state("waiting")
-        self.advertised_queues = self.client.config.get("advertised_queues")
+        self.advertised_queues = self.client.config.get(
+            "advertised_queues", {}
+        )
         self.advertised_images = self.client.config.get("advertised_images")
+        location = self.client.config.get("location")
+        if self.advertised_queues or location:
+            if not location:
+                location = ""
```

Comment:
I don't see any reason to put this check for location here. Instead, it would be much better to just handle setting it to "" if it's empty in the .get() method above